ladybug_geometry/geometry2d/arc.py

```diff
@@ -0,0 +1,400 @@
+        """
+        interval = 1 / number
+        parameter = interval
+        sub_pts = [self.p1]
+        while parameter < 1:
+            sub_pts.append(self.point_at(parameter))
+            parameter += interval
+        sub_pts.append(self.p2)
+        return sub_pts
+
+    def point_at(self, parameter):
+        """Get a point at a given fraction along the arc.
+
+        Args:
+            parameter: The fraction between the start and end point where the
+                desired point lies. For example, 0.5 will yield the midpoint.
+        """
+        _ang = self._a1 + self.angle * parameter
+        _ang = _ang if _ang <= math.pi * 2 else _ang - math.pi * 2
+        return Point2D(
+            self.c.x + math.cos(_ang) * self.r, self.c.y + math.sin(_ang) * self.r)
+
+    def point_at_angle(self, angle):
+        """Get a point at a given angle along the arc.
+
+        Args:
+            angle: The angle in radians from the start point along the arc
+                to get the Point2D.
+        """
+        _ang = self._a1 + angle
+        _ang = _ang if _ang <= math.pi * 2 else _ang - math.pi * 2
+        return Point2D(
+            self.c.x + math.cos(_ang) * self.r, self.c.y + math.sin(_ang) * self.r)
+
+    def point_at_length(self, length):
+        """Get a point at a given distance along the arc segment.
+
+        Args:
+            length: The distance along the arc from the start point where the
+                desired point lies.
+        """
+        return self.point_at(length / self.length)
+
+    def closest_point(self, point):
+        """Get the closest Point2D on this object to another Point2D.
+
+        Args:
+            point: A Point2D object to which the closest point on this object
+                will be computed.
+
+        Returns:
+            Point2D for the closest point on this line to the input point.
+        """
+        return closest_point2d_on_arc2d(point, self)
+
+    def distance_to_point(self, point):
+        """Get the minimum distance between this object and the input point.
+
+        Args:
+            point: A Point2D object to which the minimum distance will be computed.
+
+        Returns:
+            The distance to the input point.
+        """
+        close_pt = self.closest_point(point)
+        return point.distance_to_point(close_pt)
+
+    def intersect_line_ray(self, line_ray):
+        """Get the intersection between this Arc2D and another Ray2 or LineSegment2D.
+
+        Args:
+            line_ray: Another LineSegment2D or Ray2D or to intersect.
+
+        Returns:
+            A list of 2 Point2D objects if a full intersection exists.
+            A list with a single Point2D object if the line is tangent or intersects
+            only once. None if no intersection exists.
+        """
+        return intersect_line2d_arc2d(line_ray, self)
+
+    def intersect_line_ray_infinite(self, line_ray):
```

Comment:
What do you think about using `line-ray` for finite lines and `ray` for infinite cases instead of `line_ray_infinite`?

Reply:
I don't think that's a good idea because a ray only extends infinitely in one direction while this calculation uses a line extending infinitely in both directions.  I didn't put such infinite lines as their own `ladybug_geometry` objects since it is tough to represent them in most interfaces (Rhino doesn't have these infinitely extending lines, for example).  However, they are very useful for certain intersection calculations like the one here. If you want, I could change the name to `intersect_infinite_line`, which would be closer to the actual math name for what we are using (a line and not a ray or a line segment).

Reply:
I think I will just go with `intersect_line_infinite` for now. Feel free to unresolve it if you feel otherwise.